kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/SchemaManager.java

```diff
@@ -67,32 +68,90 @@ public SchemaManager(
   /**
    * Create a new table in BigQuery.
    * @param table The BigQuery table to create.
-   * @param topic The Kafka topic used to determine the schema.
+   * @param records The sink records used to determine the schema.
    */
-  public void createTable(TableId table, String topic) {
-    Schema kafkaValueSchema = schemaRetriever.retrieveSchema(table, topic, KafkaSchemaRecordType.VALUE);
-    Schema kafkaKeySchema = kafkaKeyFieldName.isPresent() ? schemaRetriever.retrieveSchema(table, topic, KafkaSchemaRecordType.KEY) : null;
-    bigQuery.create(constructTableInfo(table, kafkaKeySchema, kafkaValueSchema));
+  public void createTable(TableId table, Set<SinkRecord> records) {
+    List<com.google.cloud.bigquery.Schema> bigQuerySchemas = new ArrayList<>();
+    String tableDescription = null;
+    for (SinkRecord record: records) {
+      Schema kafkaValueSchema = schemaRetriever.retrieveValueSchema(record);
+      Schema kafkaKeySchema = kafkaKeyFieldName.isPresent() ? schemaRetriever.retrieveKeySchema(record) : null;
+      tableDescription = (kafkaValueSchema.doc() != null) ? kafkaValueSchema.doc() : tableDescription;
```

Comment:
Nit: don't need the parentheses here
```suggestion
      tableDescription = kafkaValueSchema.doc() != null ? kafkaValueSchema.doc() : tableDescription;
```